distsql/distsql.go

```diff
@@ -159,35 +160,42 @@ func canUseChunkRPC(ctx sessionctx.Context) bool {
 	if ctx.GetSessionVars().EnableStreaming {
 		return false
 	}
-	if !supportedAlignment() {
+	if !checkAlignment() {
 		return false
 	}
 	return true
 }
 
-// supportedAlignment checks the alignment in current system environment.
+var supportedAlignment = !(unsafe.Sizeof(types.MysqlTime{}) != 16 ||
+	unsafe.Sizeof(types.Time{}) != 20 ||
+	unsafe.Sizeof(types.MyDecimal{}) != 40)
+
+// checkAlignment checks the alignment in current system environment.
 // The alignment is influenced by system, machine and Golang version.
 // Using this function can guarantee the alignment is we want.
-func supportedAlignment() bool {
-	if unsafe.Sizeof(types.MysqlTime{}) != 16 ||
-		unsafe.Sizeof(types.Time{}) != 20 ||
-		unsafe.Sizeof(types.MyDecimal{}) != 40 {
-		return false
-	}
-	return true
+func checkAlignment() bool {
+	return supportedAlignment
 }
 
 // SetSystemEndian sets the system endian for the DAGRequest.
 func SetSystemEndian(dagReq *tipb.DAGRequest) {
 	dagReq.TidbSystemEndian = GetSystemEndian()
 }
 
+var systemEndian *tipb.Endian
+
 // GetSystemEndian gets the system endian.
 func GetSystemEndian() tipb.Endian {
+	if systemEndian != nil {
```

Comment:
This way is not concurrent safe, we may use `init` function to initialize the global variable `systemEndian`.